rpc/core/mempool.go

```diff
@@ -48,11 +48,12 @@ func (env *Environment) BroadcastTxSync(ctx *rpctypes.Context, tx types.Tx) (*ct
 	r := res.GetCheckTx()
 
 	return &ctypes.ResultBroadcastTx{
-		Code:      r.Code,
-		Data:      r.Data,
-		Log:       r.Log,
-		Codespace: r.Codespace,
-		Hash:      tx.Hash(),
+		Code:                r.Code,
+		Data:                r.Data,
+		Log:                 r.Log,
+		Codespace:           r.Codespace,
+		MempoolErrorMessage: r.MempoolErrorMessage,
+		Hash:                tx.Hash(),
 	}, nil
```

Comment:
We also need to set `MempoolError` during `BroadcastTxCommit` since that calls CheckTx as well (around line 107).

Reply:
I think it uses the result of `initTxCallback` directly. The code in BroadcastTxCommit calls `CheckTx` [with a callback to capture the response](https://github.com/tendermint/tendermint/blob/aacfc596993f46c0d90e43753116908094b5cb15/rpc/core/mempool.go#L40 ). This code is executed [after](https://github.com/tendermint/tendermint/blob/aacfc596993f46c0d90e43753116908094b5cb15/internal/mempool/v1/mempool.go#L287) the response is received and receives the entire response modified from `initTxCallback`. As a result, I think the field should already be set on the response we send.